sdk/internal/errorinfo/errorinfo_test.go

```diff
@@ -0,0 +1,39 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package errorinfo
+
+import (
+	"testing"
+)
+
+func TestNonRetriableError(t *testing.T) {
+	//Create sample error.
+	err := NonRetriableError(NewSampleError("Do Not Retry"))
```

Comment:
You could simplify this a bit by saying `errors.New("Do Not Retry")` and remove `NewSampleError` and friends.